src/main/kotlin/kotlinx/kover/appliers/KoverProjectApplier.kt

```diff
@@ -217,10 +218,10 @@ private fun Project.binaryReports(extension: KoverProjectConfig): List<File> {
         return emptyList()
     }
 
-    return tasks.withType(Test::class.java).asSequence()
+    return tasks.withType<Test>().asSequence()
         // task can be disabled in the project extension
         .filterNot { extension.instrumentation.excludeTasks.contains(it.name) }
-        .map { t -> t.extensions.getByType(KoverTaskExtension::class.java) }
+        .map { t -> t.extensions.getByType<KoverTaskExtension>() }
         // process binary report only from tasks with enabled cover
         .filterNot { e -> e.isDisabled.get() }
         .map { e -> e.reportFile.get().asFile }
```

Comment:
this sort of code is a good candidate for replacing with a Gradle 'Configuration' that collects the outputs of tasks, to improve compatibility with the config cache. At the moment `asSequence()` will make the tasks run immediately, rather than on request.

Reply:
The `asSequence()` itself does not launch tasks, and the call of this function is assumed only inside the provider. All this construction does not lead to the launch of tasks, other providers are designed for this.

Reply:
Sorry I meant 'be configured' instead of run. And yeah, okay, I see now that this function is only called from within a provider.

Anyway, it's not relevant to this PR, this can be revisited later.